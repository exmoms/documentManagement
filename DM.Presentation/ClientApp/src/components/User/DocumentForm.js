import React, { Component } from "react";
import Form from "react-jsonschema-form";
import "bootstrap/dist/css/bootstrap.css";

class DocumentForm extends Component {
  render() {
    const schema = this.props.schema;

    const log = (type) => console.log.bind(console, type);
    // Show the form only if the returned json is not empty
    const renderForm = () => {
      const isEmptySchema =
        Object.entries(schema).length === 0 && schema.constructor === Object;
      if (!isEmptySchema) {
        return (
          <Form
            schema={schema}
            uiSchema={this.props.uiSchema}
            formData={this.props.formData}
            onSubmit={this.props.onSubmit}
            onChange={({ formData }) => this.props.onChange(formData)}
            onError={log("errors")}
            id={this.props.id}
          >
            <br />
          </Form>
        );
      }
    };

    return <div>{renderForm()}</div>;
  }
}

export default DocumentForm;
