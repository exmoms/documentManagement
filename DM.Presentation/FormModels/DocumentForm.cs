using DM.Service.ServiceModels;
using DM.Service.ServiceModels.DocumentDTO;
using Microsoft.AspNetCore.Http;

namespace DM.Presentation.FormModels
{
    public class DocumentForm
    {
        public DocumentPostDTO Document { get; set; }
        public IFormFileCollection Scans { get ; set; }
        public IFormFileCollection Attachments { get; set; }
        public DocumentPostDTO AssignFilesFromForms()
        {
            if(Document == null)
                throw new ValidatorException();

            Document.AddAttachmentFiles(Attachments);
            Document.AddDocumentScans(Scans);

            return Document;
        }
    }
}
