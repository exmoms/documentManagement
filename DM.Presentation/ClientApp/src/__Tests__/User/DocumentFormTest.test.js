import React from "react";
import ReactDom from "react-dom";
import DocumentForm from "../../components/User/DocumentForm";
import { cleanup } from "@testing-library/react";
import { mount } from "enzyme";
import Form from "react-jsonschema-form";
import { shallow } from "enzyme";
import { act } from "react-dom/test-utils";
afterEach(cleanup);
const Testschema = {
  component: "DocumentList",
  children: [
    {
      component: "Document",
      author: "Pete Hunt1",
      children: "Document 1",
    },
    {
      component: "Document",
      author: "Pete Huntf2",
      children: "Document 2",
    },
  ],
};

const TestUiSchema = {
  component: "DocumentList",
  children: [
    {
      component: "Document",
      author: "Pete Hunt1",
      children: "Document 1",
    },
    {
      component: "Document",
      author: "Pete Huntf2",
      children: "Document 2",
    },
  ],
};

const formData = [
  {
    component: "Document",
    author: "Pete Hunt1",
    children: "Document 1",
  },
  {
    component: "Document",
    author: "Pete Huntf2",
    children: "Document 2",
  },
];

describe("DocumentForm", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    const fillFormDataMock = jest.fn();
    ReactDom.render(
      <DocumentForm
        schema={Testschema}
        uiSchema={TestUiSchema}
        formData={formData}
        onSubmit={fillFormDataMock}
        id="fill-metadata-form"
      />,
      div
    );
  });

  it("matches the snapshot", () => {
    const fillFormDataMock = jest.fn();
    const wrapper = mount(
      <DocumentForm
        schema={Testschema}
        uiSchema={TestUiSchema}
        formData={formData}
        onSubmit={fillFormDataMock}
        id="fill-metadata-form"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("Form tests of Document Form", () => {
    const fillFormDataMock = jest.fn();
    const consoleMock = jest.fn();
    console.log.bind = consoleMock;
    const wrapper = shallow(
      <DocumentForm
        schema={Testschema}
        uiSchema={TestUiSchema}
        formData={formData}
        onSubmit={fillFormDataMock}
        id="fill-metadata-form"
      />
    );
    //test the props
    expect(wrapper.find(Form).props().schema).toBe(Testschema);
    expect(wrapper.find(Form).props().uiSchema).toBe(TestUiSchema);
    expect(wrapper.find(Form).props().formData).toBe(formData);
    expect(wrapper.find(Form).props().id).toBe("fill-metadata-form");
    //test onSubmit functionality
    act(() => {
      wrapper
        .find(Form)
        .props()
        .onSubmit({ target: { value: "" } });
    });
    wrapper.update();
    expect(fillFormDataMock).toHaveBeenCalledTimes(1);
     //test onError functionality
     act(() => {
      wrapper
        .find(Form)
        .props()
        .onError;
    });
    wrapper.update();
    expect(consoleMock).toHaveBeenCalledTimes(1);
    expect(consoleMock.mock.calls[0][0]).toBe(console);
    expect(consoleMock.mock.calls[0][1]).toBe("errors");

  });
});
