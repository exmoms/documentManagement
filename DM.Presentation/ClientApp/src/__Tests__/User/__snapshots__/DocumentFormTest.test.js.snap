// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentForm matches the snapshot 1`] = `ReactWrapper {}`;
